net/query_filter/query_filter.h

```diff
@@ -0,0 +1,27 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_NET_QUERY_FILTER_QUERY_FILTER_H_
+#define BRAVE_NET_QUERY_FILTER_QUERY_FILTER_H_
+
+#include "net/base/net_export.h"
+#include "third_party/abseil-cpp/absl/types/optional.h"
+#include "url/origin.h"
+
+class GURL;
+
+namespace net::query_filter {
+
+absl::optional<GURL> ApplyQueryFilter(const GURL& original_url);
```

Comment:
1. `NET_EXPORT`. I think it currently works without `NET_EXPORT`, because you use `//brave/net:query_filter` target, which you should not do.
2. please add comments to these functions.